jsx/StaticDataTable.js

```diff
@@ -193,14 +267,8 @@ StaticDataTable = React.createClass({
                     data = this.props.Data[index[i].RowIdx][j];
                 }
 
-                // Increase counter, if filter value is found to be a substring
-                // of one of the column values
-                var filterData = this.props.Filter[this.props.Headers[j]];
-                if (filterData !== null &&
-                    data !== null &&
-                    data.indexOf(filterData) > -1
-                ) {
-                    filterMatchCount++;
+                if (this.hasFilterKeyword(this.props.Headers[j], data)) {
```

Comment:
@driusan There are 2 loops, outside loop iterates through rows of data, inside loop iterates through columns of a specific row. filterMatchCount counts how many `Filter` properties are matched in the row of data.

On first iteration, we check if Filter contains property `Name` (first column). Since it does, and the data is the same in filter and column, we increase filterMatchCount by 1. 

On second iteration, we check if Filter contains property `State` (second column). 
Filter does not contain that property. So `Filter['State']` will evaluate to `null` and `null !== 'CaDAVEr'`. So filterMatchCount remains 1. Since Filter contains only 1 property (key/value pair) we know that this row should be displayed in the results (filterMatchCount == Filter length / 1 == 1)

For the next iteration of outside loop filterMatchCount is reset to 0, and the process it repeated.

If it's still not clear, I should probably go over the code with you offline.